week5.ipynb

```diff
@@ -0,0 +1,351 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A note on reference frames"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A reference frame defines the physical space that we wish to consider within a given problem. Imagine a ball sitting at the top of a 10 foot tall flight of stairs. If we want to use the stairwell as our reference frame, then the initial height of the ball would be 10 feet, and it could fall down the stairs to land at a final height of 0 feet. But what if those stairs start on the sixth floor of a building (that is, at 60 feet above the ground) and climb to the seventh floor (70 feet above the ground)? By casting the problem this way, we've changed our reference frame to be that of the earth, where the ground below our feet is taken to be 0. You can see how choosing a frame of reference is important in calculations of potential energy; our convention in this assignment will be to treat the ground as 0 for all problems."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## An example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's imagine an apple falling to the ground from a height of 10 feet. Its mass is 100 g. We will calculate how much work was done on the apple, and in the 'Questions' section below you will find the change in velocity that it experienced during its fall. We will assume the apple was stationary immediately before falling."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# this exercise only requires elementary math which can all be done in basic R, \n",
+    "# so we don't need to import any additional packages\n",
+    "\n",
+    "# in the questions below you will need to take a square root\n",
+    "# this can be done by typing sqrt(...)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# before we do anything else, we should convert units! Convert all to base SI units\n",
+    "\n",
+    "height_initial <- 10 / 3.3 # 3.3 feet in 1 meter\n",
+    "mass <- 100 /1000 # 1000 grams to a kilogram"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# now we can create a list of constants\n",
+    "\n",
+    "constants <- list('mass_of_apple' = mass, \n",
+    "                  'gravitational_acceleration' = 9.81, \n",
+    "                  'initial_height' = height_initial, \n",
+    "                  'final_height' = 0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# calculate potential energy\n",
+    "\n",
+    "potential_energy = constants[[1]] * constants[[2]] * (constants[[4]] - constants[[3]])\n",
```

Comment:
It's interesting to see the calculation done this way in R. I've always done it with a dollar sign like constants$gravitational_acceleration where you call the variable list and select a specific value in that list.